contracts/src/v0.8/l2ep/dev/zksync/ZKSyncSequencerUptimeFeed.sol

```diff
@@ -0,0 +1,272 @@
+   * @dev This function will revert if not called from `l1Sender` via the L1->L2 messenger.
+   *
+   * @param status Sequencer status
+   * @param timestamp Block timestamp of status update
+   */
+  function updateStatus(bool status, uint64 timestamp) external override {
+    // TODO: Address aliasing will be necessary to check on the L2 the L1 caller. 
+    // The aliasing can be applied using AddressAliasHelper.applyL1ToL2Alias(msg.sender).
+
+    FeedState memory feedState = s_feedState;
+    if (
+      msg.sender != address(s_l2SharedBridge) || s_l2SharedBridge.l1Bridge() != s_l1Sender
+    ) {
+      revert InvalidSender();
+    }
+
+    // Ignore if latest recorded timestamp is newer
+    if (feedState.startedAt > timestamp) {
+      emit UpdateIgnored(feedState.latestStatus, feedState.startedAt, status, timestamp);
+      return;
+    }
+
+    if (feedState.latestStatus == status) {
+      _updateRound(feedState.latestRoundId, status);
+    } else {
+      feedState.latestRoundId += 1;
+      _recordRound(feedState.latestRoundId, status, timestamp);
+    }
+  }
+
+  /// @inheritdoc AggregatorInterface
+  function latestAnswer() external view override checkAccess returns (int256) {
+    FeedState memory feedState = s_feedState;
+    return _getStatusAnswer(feedState.latestStatus);
+  }
+
+  /// @inheritdoc AggregatorInterface
+  function latestTimestamp() external view override checkAccess returns (uint256) {
+    FeedState memory feedState = s_feedState;
+    return feedState.startedAt;
+  }
+
+  /// @inheritdoc AggregatorInterface
+  function latestRound() external view override checkAccess returns (uint256) {
+    FeedState memory feedState = s_feedState;
+    return feedState.latestRoundId;
+  }
+
+  /// @inheritdoc AggregatorInterface
+  function getAnswer(uint256 roundId) external view override checkAccess returns (int256) {
+    if (_isValidRound(roundId)) {
+      return _getStatusAnswer(s_rounds[uint80(roundId)].status);
+    }
+
+    revert NoDataPresent();
+  }
+
+  /// @inheritdoc AggregatorInterface
+  function getTimestamp(uint256 roundId) external view override checkAccess returns (uint256) {
+    if (_isValidRound(roundId)) {
+      return s_rounds[uint80(roundId)].startedAt;
+    }
+
+    revert NoDataPresent();
+  }
+
+  /// @inheritdoc AggregatorV3Interface
+  function getRoundData(
+    uint80 _roundId
+  )
+    public
+    view
+    override
+    checkAccess
+    returns (uint80 roundId, int256 answer, uint256 startedAt, uint256 updatedAt, uint80 answeredInRound)
+  {
+    if (_isValidRound(_roundId)) {
+      Round memory round = s_rounds[_roundId];
+      answer = _getStatusAnswer(round.status);
+      startedAt = uint256(round.startedAt);
+      roundId = _roundId;
+      updatedAt = uint256(round.updatedAt);
+      answeredInRound = roundId;
+    } else {
+      revert NoDataPresent();
+    }
+    return (roundId, answer, startedAt, updatedAt, answeredInRound);
```

Comment:
Small gas optimization tip - let's avoid reassigning the variables here since they are only used once:

```suggestion
   if (!_isValidRound(_roundId)) {
        revert NoDataPresent();
    }

    Round memory round = s_rounds[_roundId];

    return (_roundId, _getStatusAnswer(round.status), uint256(round.startedAt), uint256(round.updatedAt), _roundId);
```